synthetic/image/poisson_blending.py

```diff
@@ -15,77 +15,71 @@
     return mask
 
 
 # the below poisson blending function is courtesy of
 # https://github.com/parosky/poissonblending/blob/master/poissonblending.py
 def blend(image_target, image_source, image_mask, offset=(0, 0)):
-    region_source = (
-        max(-offset[0], 0),
-        max(-offset[1], 0),
-        min(image_target.shape[0] - offset[0], image_source.shape[0]),
-        min(image_target.shape[1] - offset[1], image_source.shape[1]),
-    )
-    region_target = (
-        max(offset[0], 0),
-        max(offset[1], 0),
-        min(image_target.shape[0], image_source.shape[0] + offset[0]),
-        min(image_target.shape[1], image_source.shape[1] + offset[1]),
-    )
+    offset_x, offset_y = offset
+
+    region_source_xmin = max(-offset_x, 0)
+    region_source_ymin = max(-offset_y, 0)
+    region_source_xmax = min(image_target.shape[0] - offset_x, image_source.shape[0])
+    region_source_ymax = min(image_target.shape[1] - offset_y, image_source.shape[1])
+
+    region_target_xmin = max(offset_x, 0)
+    region_target_ymin = max(offset_y, 0)
+    region_target_xmax = min(image_target.shape[0], image_source.shape[0] + offset_x)
+    region_target_ymax = min(image_target.shape[1], image_source.shape[1] + offset_y)
+
     region_size = (
-        region_source[2] - region_source[0],
-        region_source[3] - region_source[1],
+        region_source_xmax - region_source_xmin,
+        region_source_ymax - region_source_ymin,
     )
 
-    image_mask = image_mask[region_source[0]:region_source[2], region_source[1]:region_source[3]]
+    image_mask = image_mask[region_source_xmin:region_source_xmax, region_source_ymin:region_source_ymax]
     image_mask = prepare_mask(image_mask)
     image_mask[image_mask == 0] = False
     image_mask[image_mask != False] = True
 
     A = scipy.sparse.identity(np.prod(region_size), format='lil')
-    for y in range(region_size[0]):
-        for x in range(region_size[1]):
-            if image_mask[y, x]:
-                index = x + y * region_size[1]
-                A[index, index] = 4
-                if index+1 < np.prod(region_size):
-                    A[index, index + 1] = -1
-                if index-1 >= 0:
-                    A[index, index - 1] = -1
-                if index+region_size[1] < np.prod(region_size):
-                    A[index, index + region_size[1]] = -1
-                if index-region_size[1] >= 0:
-                    A[index, index - region_size[1]] = -1
+    for y, x in itertools.product(range(region_size[0]), range(region_size[1])):
+        if image_mask[y, x]:
+            index = x + y * region_size[1]
+            A[index, index] = 4
+            if index + 1 < np.prod(region_size):
+                A[index, index + 1] = -1
+            if index - 1 >= 0:
+                A[index, index - 1] = -1
+            if index+region_size[1] < np.prod(region_size):
+                A[index, index + region_size[1]] = -1
+            if index-region_size[1] >= 0:
```

Comment:
Give the operators some space for breathing
```suggestion
            if index + region_size[1] < np.prod(region_size):
                A[index, index + region_size[1]] = -1
            if index - region_size[1] >= 0:
```